Sources/Datadog/Tracing/Utils/ActiveSpanUtils.swift

```diff
@@ -0,0 +1,45 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+import os.activity
+
+/// Helper class to get the current Span
+internal struct ActiveSpanUtils {
```

Comment:
How about making it a real class and assigning this object to a `Tracer`, i.e.:
```swift
public class Tracer: OTTracer {
   // ...
   internal let activeSpansPool = ActiveSpansPool()
}
```
?

This way we won't need to hassle with `XCTAssertNil(ActiveSpanUtils.getActiveSpan())` checks in unit tests, as the pool will be local to the `Tracer` and its lifecycle will follow the tracer's lifecycle (which is already enforced in tests). I should be less error-prone and produce less flakiness.

For further `XCTests` auto instrumentation we could either use `Global.sharedTracer` to issue `XCTest` spans OR if we decide to use a separate instance of `Tracer`, we could pass it the `Global.sharedTracer.activeSpansPool` on initialization.

Reply:
Great idea, much better than it is now, implementing now